test/test.js

```diff
@@ -1,27 +1,50 @@
 function readBlobAsText(blob) {
   return new Promise(function(resolve, reject) {
```

Comment:
We won't need any promise wrapping in the sync case, you can just define the function as

``` js
function readBlobAsText(blob) {
  return new FileReaderSync().readAsText(blob);
}
```